http-clients/src/main/java/com/palantir/remoting/http/errors/SerializableErrorErrorDecoder.java

```diff
@@ -101,18 +106,16 @@ public Exception decode(String methodKey, Response response) {
         }
     }
 
-    private static Exception constructException(Class<? extends Exception> exceptionClass, String message) {
-        try {
-            return exceptionClass.getConstructor(String.class).newInstance(message);
-        } catch (Exception e1) {
-            return new RuntimeException(String.format(
-                    "Failed to construction exception of type %s, constructing RuntimeException instead: %s%n%s",
-                    exceptionClass.toString(), e1.toString(), message));
+    private static Exception constructException(
```

Comment:
Hey Jason! HTTP-remoting's handling of exceptions was actually doubly broken:
- most exceptions were thrown as 500s on server side (what you're mentioning)
- most exceptions were failing to decode on client side (what this commit is)

Both of these errors were related to the same failure to deal with WebApplication, ClientError, or ServerError exceptions, just in different places. I already fixed the problem you mention at https://github.com/palantir/http-remoting/pull/48 .